src/ActionParameterHandler.ts

```diff
@@ -125,6 +125,13 @@ export class ActionParameterHandler {
 
                 return !!value;
 
+            case "date":
+                const parsedDate = new Date(value);
+                if (isNaN(parsedDate.getTime())) {
+                    throw new BadRequestError(`${param.name} is invalid! It can be parsed to date.`);
```

Comment:
I didn't like new try/catch block in next file. Can you try here do `Promise.reject(new BadRequestError(`${param.name} is invalid! It can be parsed to date.`))`; instead of ` throw new BadRequestError(`${param.name} is invalid! It can be parsed to date.`);` ? I think it should work without having to add extra try catch block